### DISCUSSION TITLE
ImportError (mac 개발환경 구축중)

### DISCUSSION THREAD

**Discussion by min1826** [lablup/contribution-academy-2021#56] [Q&A]
### Describe the problem
macOS 환경에서 install-dev.sh로 개발환경 구축 도중 Setting up databases에서 ImportError 에러가 발생합니다.

`ImportError: dlopen(): Symbol not found: ____chkstk_darwin`

### To Reproduce
brew install python3
brew install pyenv
brew install pyenv-virtualenv
bash install-dev.sh 실행 도중 ImportError 에러가 발생합니다.
install-dev.sh에서 에러가 발생하는 부분은 `python -m ai.backend.manager.cli schema oneshot` 인 것 같습니다.
(`python -m ai.backend.manager.cli schema oneshot` 을 실행했을 때 같은 오류가 발생합니다.)

* install-dev.sh에서 에러가 발생하는 부분
```
### DB schema
show_info "Setting up databases..."
cd "${INSTALL_PATH}/manager"
python -m ai.backend.manager.cli schema oneshot
python -m ai.backend.manager.cli fixture populate fixtures/example-keypairs.json
python -m ai.backend.manager.cli fixture populate fixtures/example-resource-presets.json
```

* Error Message
```
ImportError: dlopen(/Users/minkyeong/.pyenv/versions/3.9.5/envs/venv-537gg3ly-manager/lib/python3.9/site-packages/psycopg2/_psycopg.cpython-39-darwin.so, 2): Symbol not found: ____chkstk_darwin
Referenced from: /Users/minkyeong/.pyenv/versions/3.9.5/envs/venv-537gg3ly-manager/lib/python3.9/site-packages/psycopg2/.dylibs/libcrypto.1.1.dylib (which was built for Mac OS X 10.15)
Expected in: /usr/lib/libSystem.B.dylib
in /Users/minkyeong/.pyenv/versions/3.9.5/envs/venv-537gg3ly-manager/lib/python3.9/site-packages/psycopg2/.dylibs/libcrypto.1.1.dylib
```

### Expected behavior
개발 환경 구축이 정상적으로 완료되어야합니다.

### Screenshots
* `bash install-dev.sh` 실행시켰을 때
![1](https://user-images.githubusercontent.com/71456233/129572132-4097f570-6d13-42af-a428-68c4f2faea67.png)

* `python -m ai.backend.manager.cli schema oneshot` 만 따로 실행시켰을 때
![2](https://user-images.githubusercontent.com/71456233/129572173-f149741c-4ec0-49a7-8ab7-300cbd1f1b31.png)

### Environment Description
OS: macOS High Sierra 10.13.6
Current Shell: bash

![3](https://user-images.githubusercontent.com/71456233/129572157-dd07383a-cbb6-4535-ae0c-25083a833ec0.png)

**Comment by lizable** [comment]
How about installing xcode command line tool and restart the installation?
Here's the [link](https://stackoverflow.com/questions/23015152/git-command-line-on-mac-os-error-dyld-symbol-not-found-strlcpy-chk) that might give a clue.(please check Jason's Answer!)
Btw, Do you have any plans to upgrade OS version? It's essential for preventing any security issues via patches.

**Reply by min1826** [reply]
I've installed xcode command line tool and restarted the installation, but same error occurred.
![2](https://user-images.githubusercontent.com/71456233/129687100-ee9d0fb5-7dd9-491d-a8b3-1e354c04d8c5.png)
![image](https://user-images.githubusercontent.com/71456233/129685625-76a16c39-98fa-43f1-8f56-00b61f5ecf6a.png)

And 10.13.6 is the latest version I can update...

**Reply by lizable** [reply]
@min1826 Check this [link](https://stackoverflow.com/questions/27264574/import-psycopg2-library-not-loaded-libssl-1-0-0-dylib) and follow the NSTJ's answer.
